The big one: Gang Wars - Gangpocalypse Edition

Robustin opened the pull request:
Ohhhh shit:

- Fuckin Vigilantes
- Vigilante mechanics and items
- New weapons
- Tweaks and shit
- Some fixes too
- Secs gone
- Armory's gone
- AI's gone
- Reinforcements (with advanced anti-snowball MATH) are available
- Reinforcement use a special ghost poll that is more selective than usual, it will ask dead gangsters from the same gang and if nobody responds then it will ask dead vigilantes, dead gangsters cannot be reinforcements for a rival gang. 
- You can revive bros
Credit to Cyberboss for blowing up the brig and disabling jobs code, credit to FantasticFwoosh for the teleporter icon.

Cliffnotes:

Of the things people don't like about Gang War some of the most common complaints were how the round would drag on despite its incredibly high deathcount, how it really sucks getting taken out early in the round (its like rev except now your opponents are less likely to deconvert you than they are to just kill you and space your corpse), and how security just sucks the life out of the mode by either rolling gangsters with a flood of implants or failling to do so and having the first gang to get sec access subsequently rob the armory and steamroll the other gangs. 

My other PR's have touched on some of this. Decentralizing points has only been around for a few days and makes gang members more in control of their own fate, this is the final of the BIG changes and should synergize with previous changes to make GW into a truly unique mode that will have a place in our rotation.

Robustin committed:
Fixing the master

Robustin committed:
Update tgstation.dme

Robustin committed:
Vigilante Work

Robustin committed:
Almost done

Robustin committed:
Framework finished

Robustin committed:
Da big wun

Robustin committed:
Merge branch 'Future9' of https://github.com/Robustin/tgstation into Vigilante

Steelpoint commented:
Would you care to explain some of the points.

More so the points about 'vigilante mechanics', Reinforcements' and 'reviving people'.

Robustin committed:
==

RemieRichards commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -92,6 +113,83 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
+
+/datum/game_mode/gang/proc/gangpocalypse()
+	set waitfor = FALSE
+	..(FALSE, FALSE)
```
this proc has no parent!

Cyberboss replied:
Copypasta of my code detected

Robustin replied:
Guilty

Robustin commented:
I'm just going to push for a testmerge soon and hopefully that will make it clear.

Reinforcements and Revives are ways for gangs to bring their dead back into the round, but both have very significant anti-snowball mechanics built in. If your gang is in dead-last then reinforcements are pretty much the only way you can stage a comeback. 

Revive is a reliable back-from-the-dead injector that scales up in cost depending on the size of your gang. It's available to all gang members and the revived individuals are brought back with most of their wounds healed, but a significant amount of brain damage.

Reinforcements run on a pretty complex formula I put together. The price can range from "nearly free" to "obscenely expensive" depending on the relative strength of your gang. Furthermore, the frequency of reinforcements scales even more drastically depending on the relative strength of your gang. A dominant gang will only see a reinforcement or two every 10 minutes while a gang with only a couple survivors can see 5-8x that amount. 

Reinforcements come in as randomly generated assistants wearing the gang's outfit and possessing a surplus rifle, switchblade, and an internal tool with just enough points for a spraycan. I honestly have no idea if the reinforcements will be negligible or decisive in how they alter rounds. The only upside is that if one sides starts to get the upper hand then their reinforcement flow will slow down and the opponents will pick up.

However the reinforcements come from a teleporter machine and it can be destroyed like anything else.

That reminds me I need to put a territory requirement for that thing.

FantasticFwoosh commented:
![](https://i.ytimg.com/vi/iJWqzQYh4k0/hqdefault.jpg)

Steelpoint commented:
I think the one benefit I can see from these overhauls is that it simply removes any pretense and just turns everything in the game towards the benefit of a team vs team vs team conflict. 

Time will tell if this works well.

Robustin commented:
@TDM Memesters

Look at the top 3 modes that don't start with "Traitor".

Nukeops, Revolution, and Wizard.

These are all TDM modes. Yes even wizard, who either summons shit and it becomes the lowest fucking tier of TDM imaginable, or the wizard tries to murderbone and you get a few minutes of TDM before it reverts to traitor. 

Trying to make a mode called Gang WAR into anything other than TDM is destined for mediocrity. If Gang War can create some top-tier gameplay then maybe it can swap weights with that dumpster fire of a mode we called Wizard.

BeeSting12 commented:
Wizard is shit, I agree. But see my post on the forums in the gang thread for my opinion- the summary is that it feels like you're turning the station into a map for a team death match rather than a place where people do their jobs while antagonists work against them. I think that, knowing the players, they'd pretty much drop what they're doing and run around powergaming with little to no roleplay about it. I doubt any science, engineering, or medical would get done at all. But that's just my 2 cents for what it's worth, and I still think its worth a testmerge

Robustin commented:
Medical will definitely be getting done, it will be an interesting turf war since the vastly increased weaponization of gangs and no security means each side will need much more medical attention and can be far more assertive in trying to get it. 

I understand this mode is disruptive and it shouldn't be a mainstay that makes up 20% of our rounds, but Summon Wizards and Warops are even more disruptive and I would gladly take up round weighting from the former at least. 

I just appreciate variety and I think this mode provides it and in time people will agree that it deserves a spot in rotation.

BeeSting12 commented:
I think you've done a good job balancing this so far so I'd like to see how this plays out. At very least it'll be an interesting learning experience.

kevinz000 commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -39,6 +41,25 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
+				break
+
+	if(!target_armory || !target_equipment)
+		message_admins("No armory/equipment room detected, unable to start gang!")
```
... why can't you start without a brig?

Robustin replied:
This was Cyber's solution instead of having to manually delete everything in the brig.

I will have to tailor some other solutions potentially for TTV's, Cap's ID, etc.

Cyberboss replied:
If a map doesnt have sec, it's an error

kevinz000 commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -267,20 +460,74 @@
 	item_path = /obj/item/ammo_box/magazine/uzim9mm
 
+/datum/gang_item/weapon/launcher
+	name = "88mm high explosive rocket launcher"
```
And here I thought this would be balanced.

Robustin replied:
Its a sleeper UP item. Basically a ranged welderbomb (wider radius but only light explosion) that can only detonate on dense objects. Also 3 seconds to aim. We'll see though.

kevinz000 commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -312,36 +568,147 @@
+
+////// Gangbuster Seraph //////
+
+/datum/gang_item/equipment/seraph
```
Here I thought it would be balanced!

Robustin replied:
We discussed this but youre shooting ballistics against gangsters fully armored against ballistics in basically a reskinned durand with a couple extra whistles.

kevinz000 commented:
:+1: I'm going to enjoy being security only even more powerful.

Steelpoint commented:
So how do vigilantes work, do you spawn as one, do you opt in or does the entire crew start as one but they can chose to stay or go gang banger?

Cyberboss commented on code/game/objects/items/weapons/clown_items.dm:
```diff
@@ -41,6 +41,17 @@
+/obj/item/weapon/soap/vigilante/Crossed(AM as mob|obj)
+	if (istype(AM, /mob/living/carbon))
+		var/mob/living/carbon/M = AM
+		M.slip(2, 2, src)
```
Make slipping a common proc on soap and override it

Robustin replied:
I think a var makes more sense here actually

Cyberboss commented on code/game/objects/items/weapons/implants/implant_gang.dm:
```diff
@@ -32,17 +32,20 @@
-			if(SSticker.mode.remove_gangster(target.mind,0,1))
-				success = 1	//Was not a gang boss, convert as usual
+			if(SSticker.mode.remove_gangster(target.mind))
+				success = 1     //Was not a gang boss, convert as usual
```
TRE

Tacolizard replied:
TRE

Cyberboss commented on code/game/objects/items/weapons/implants/implant_gang.dm:
```diff
@@ -32,17 +32,20 @@
+				var/datum/gang/G = user.mind.gang_datum
+				SSticker.mode.add_gangster(target.mind,G)
+				target.Paralyse(3)
+				return 1
```
TRUE

Cyberboss commented on code/game/objects/items/weapons/implants/implant_gang.dm:
```diff
@@ -32,17 +32,20 @@
-
-		qdel(src)
-		return 0
+				return 1
```
TRUE

Robustin commented on code/game/gamemodes/gang/vigilante.dm:
```diff
@@ -0,0 +1,203 @@
+		/datum/gang_item/equipment/seraph
+		)
+
+/obj/item/device/vigilante_tool/New(mob/user)
```
Then Remie or whoever goes "nah this should be a static list" AHHH.

Cyberboss replied:
What?? I'm talking about the NEW!~!!!!

Robustin replied:
ohhh

Robustin commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -332,4 +438,4 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
 			SSticker.mode.explosion_in_progress = 0
 			SSticker.force_ending = TRUE
-
+			SSticker.mode.auto_declare_completion_gang(G)
```
Nope, hence why the post-round credits aren't printing atm.

evsey9 commented:
Praise communism

Robustin commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -92,6 +113,83 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
+		if(!is_gangster(M))
+			vigilize(M)
+
+/proc/gangtest()
```
its a debug proc since I can't get the actual mode running locally no matter how much I mess with the reqs/debug.

Robustin commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -92,6 +113,83 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
+	explosion(target_armory, 10, 16, 28, 30, TRUE, TRUE)
+	explosion(target_equipment, 10, 16, 28, 30, TRUE, TRUE)
+	for(var/mob/living/M in GLOB.player_list)
+		vigilize(M)
```
Are you saying I should limit it to station z_level, or is there a better way to filter out non-crew?

Robustin commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -92,6 +113,83 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
+	M.mind.objectives += E
+	to_chat(M, "<FONT size=3><u><b>You are a Vigilante!</b></u><br> Nanotrasen has given all loyal crew the authority to eliminate gang activity aboard the station.<br> You possess a reverse-engineered gangtool that rewards influence for destroying gangster equipment.<br> You will also receive influence for keeping the station free of gang tags.<br><b>Prevent gangs from taking over the station!<b></ ...
+	M.mind.announce_objectives()
+	new /obj/item/device/vigilante_tool(M)
```
Works during testing

Robustin commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -186,19 +284,27 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
 	gangster_mind.special_role = "[G.name] Gangster"
-
+	for(var/obj/O in gangster_mind.current.contents)
+		if(istype(O, /obj/item/device/vigilante_tool))
```
right I keep forgetting that it already filters out non-types.

Robustin commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -75,6 +76,129 @@
+	spawn_gangster()
+	if(we == 0)
+		we = 1
+	cooldown = 200+((we/(rival+we))*50)**2
```
?

Cyberboss replied:
200 and 50 should be sefines

Robustin replied:
Could you clarify why? Should it be really descriptive defines so people might understand it better because these numbers only appear once and to be honest they ARE just mostly arbitrary numbers I plugged into a spreadsheet until I got results I liked. 200 is the floor that helps makes the differences in spawn rates less drastic and 50 is just the multiplier that gets me reasonable numbers.

Robustin commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -75,6 +76,129 @@
+	var/obj/item/clothing/suit/outerwear = new G.outer_outfit(H)
+	outerwear.armor = list(melee = 20, bullet = 35, laser = 10, energy = 10, bomb = 30, bio = 0, rad = 0, fire = 30, acid = 30)
+	outerwear.body_parts_covered = CHEST|GROIN|LEGS|ARMS
+	outerwear.desc += " Tailored for the [G.name] Gang to offer the wearer moderate protection against ballistics and physical trauma."
```
how can I make a subtype that covers like 8 different types of clothing?

Make a new type and then make 8 new subtypes for it? Seems like even more confusion/work

Robustin commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -75,6 +76,129 @@
+	H.equip_to_slot_or_del(new /obj/item/clothing/shoes/jackboots(H), slot_shoes)
+	H.put_in_l_hand(new /obj/item/weapon/gun/ballistic/automatic/surplus/gang(H))
+	H.equip_to_slot_or_del(new /obj/item/ammo_box/magazine/m10mm/rifle(H),slot_l_store)
+	H.equip_to_slot_or_del(new /obj/item/weapon/switchblade(H),slot_r_store)
```
I tried, problem is it relies on the gang's unique clothing and I couldn't find a way to get that information in the outfit datum.

Cyberboss replied:
Override outfit/New to set the variables to what you want based on parameters

Cyberboss replied:
```
/datum/outfit/gang
   //static stuff here

/datum/outfit/gang/New(uniform, outwear, etc...)
   //set outfit vars to params
```

Robustin commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -312,36 +568,147 @@
+			H.emote("gasp")
+			H.setBrainLoss(70)
+			add_logs(user, H, "revived", src)
+			qdel(src)
```
I wasn't really impressed with the gang implanter's version that leaves behind a bunch of needles but I suppose its worth doing.

Robustin commented on code/game/gamemodes/gang/vigilante.dm:
```diff
@@ -0,0 +1,203 @@
+			Cat += G
+		else
+			vig_category_list[G.category] = list(G)
+	addtimer(CALLBACK(src, .proc/income), 1800)
```
literally never seen that before, ill look into it but feel free to explain further.

Cyberboss replied:
It prevents the same timer from being created twice, since this is a recurring action you need it

Robustin commented on code/game/gamemodes/gang/vigilante.dm:
```diff
@@ -0,0 +1,203 @@
+		all_territory += G.territory
+	for(var/area/A in tags)
+		if(A in all_territory)
+		else
```
Credits the vigilante for territories they've cleaned, which remain clean (aka aren't in any gang's territory list).

Cyberboss replied:
so just do if(!(A in all_territory)) instead of this dumb control flow

Robustin replied:
Ah yea that was a byproduct of me debugging, couldnt figure out the issue until I realized the tag's territory var and the actual area were slightly different because of how they were being defined - solved that so I can move it back to where I had it before.

Robustin commented on code/game/gamemodes/gang/vigilante.dm:
```diff
@@ -0,0 +1,203 @@
+	if(!I)
+		to_chat(user, "<span class='notice'> No item detected")
+		return
+	switch(I.type)
```
It works as intended, so how should I be doing it?

Cyberboss replied:
it will only work on EXACT type matches, so any subtypes in the future won't trigger it. @RemieRichards can tell you the right way

Robustin commented:
@Steelpoint 

Everyone who isn't selected to be a boss started as a vigilante. How you play that role is up to you. You can go full Kevinz and basically try to play one-man security. You can be an opportunist and try to disarm a conversion pen or push a distracted boss and cash in their shit for a decent gun. Or you can mind your own business and try to stay out of it. 

Though technically vigilantes will "greentext" if they escape alive.

Bawhoppen commented:
Just because other modes are TDM, doesn't make an excuse for this one to be. Don't forget how traitor is basically universally acclaimed as the best mode, and the most played.

Robustin committed:
quick fixes

Robustin committed:
Merge branch 'Future9' of https://github.com/Robustin/tgstation into Vigilante

Robustin commented:
I have mixed feelings about traitor, especially about how stale its gameplay has been for so many years, but I understand why its the "core" mode, every round cannot be a deathmatch and I don't want it that way. I would much rather GW take a chunk out of wizard - where the best thing you can say about wizard rounds is "hey at least its pretty much a traitor round after 5 minutes". 

Gang War was always TDM though, this is just about seeing if we can't embrace that in a mode that has "WAR" in the name. I don't think anyone would mind if this took a point or two out of rev or nukeops anyway.  I'm just pushing back at people who seem to think this is our first TDM mode when it literally defines the majority of our non-traitor lineup.

PKPenguin321 commented:
Tdm is a stupid buzzword since it implies there's no objective but murder

Tacolizard commented:
>Ohhhh shit:

    

>- Fuckin Vigilantes


now that, ladies and gentlemen, is how you start a PR

Tacolizard commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -75,6 +76,129 @@
+	if(we == 0)
+		we = 1
+	cooldown = 200+((we/(rival+we))*50)**2
+	world << "[cooldown] deciseconds"
```
delet this

Tacolizard commented:
im HYPED for this PR

bgobandit commented:
I could not possibly be more against the idea of getting rid of roles or station areas just to prop up yet another TDM mode

Bawhoppen commented:
^

Robustin commented:
Vigilantes can play the role of security without turning cargo into a concentration camp every round. If you see roles as something deeper than "this is my title and the color of my jumpsuit" then you shouldn't be concerned. 

I honestly can't fathom the reasoning behind people that would want to prop up Wizard over Gang War. Wizard is "not SS13" in many ways. It doesn't match the setting or the core themes (0 paranoia outside of the occasional mindswap meme) and roles go out the fucking window when the HOS is murdering the brig for guns and everyone mindswapped into another job. Yet miraculously the mode is in our Top 5 simply by virtue of being around forever and people just accepting that shitty wizard rounds are something we have to deal with. 

War operatives were the same concept. We took a mode with conflicting themes (Should Ops be a bunch of slick operators who try to slip in and out unnoticed or should they always go in loud so that there's actually a conflict during the round?) and we opted to try the LOUD and VIOLENT approach where we gave Ops the chance to shed any pretense that they are secret or stealthy and just have a 10 minute TDM round preceded by 20 minutes of prep. Yet I don't think anyone can say the RAW CARNAGE AND DEATH approach wasn't an improvement for Nukeops. We gave them more toys too and the combination brought new life to a pretty stagnant mode. 

Again I'm not arguing that this should be the ideal for every round - but I don't think anyone can argue that the occasional round of intense combat between two (or more) teams without any pretense of normalcy isn't fun and healthy after 6 rounds of "M-M-MURDER TRAITOR M-M-MAINTENANCE!!!"

FantasticFwoosh commented:
@bgobandit the way i see it it's a evolution of our existing antag types because usually we factor in some sort of in station counter, and how the antag fits around station life etc.

FantasticFwoosh commented:
@bgobandit the way i see it it's a evolution of our existing antag types because usually we factor in some sort of in station counter, and how the antag fits around station life etc.

Here is throwing out the rulebook for good purpose, the same as war ops and highlander does to a extent but since sec is too oppressive a counter the antags are set against each other.

bgobandit commented:
War ops fits into the existing station/game structure. Highlander is an admin button that is reserved for memes, and even that fits into the existing station structure.

Robustin committed:
Merge branch 'master' into Future9

Robustin commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -92,6 +113,83 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
+	for(var/mob/living/M in GLOB.player_list)
+		vigilize(M)
+
+/proc/vigilize(mob/living/M)
```
I will, I made it global for debugging since I can't get Gang War to start during testing because of other checks that would be more dangerous to remove for testing purposes.

Robustin commented on code/game/gamemodes/gang/vigilante.dm:
```diff
@@ -0,0 +1,203 @@
+
+/obj/item/device/vigilante_tool/New(mob/user)
+	var/datum/action/innate/vigilante_tool/VT = new
+	VT.Grant(user, src)
```
You mean like a var for "linked action" or whatever?

Robustin committed:
Update gang.dm

Robustin committed:
Update vigilante.dm

Robustin committed:
Update

Robustin commented:
Warops don't really do that at all though. All-access goes out and the station's job structure immediately loses all coherence, the only people who are distinguished are the disk holder and the pinpointer holder. Everyone else is Captain Jr.

bgobandit commented:
The difference is it does not suddenly remove large swaths of the game, the players have to do that. I'm not a fan of that dynamic but it is not hardcoded

Robustin commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -75,6 +76,129 @@
+	density = TRUE
+	icon = 'icons/obj/machines/teleporter.dmi'
+	icon_state = "gang_teleporter_on"
+	var/datum/gang/G
```
theres already so many variation of gang in this code and we use G in other gang procs

Robustin commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -75,6 +76,129 @@
+	icon = 'icons/obj/machines/teleporter.dmi'
+	icon_state = "gang_teleporter_on"
+	var/datum/gang/G
+	var/list/queue = list()
```
I cannot see anywhere in our code where a list is created by LAZYLIST

Robustin commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -312,36 +568,147 @@
+
+/obj/item/weapon/reviver/attack(mob/living/carbon/human/H, mob/user)
+	if(!ishuman(H))
+		return
```
I'm gonna wait on this. I really don't like the expansion of intents where they don't serve a purpose. This only has one purpose, healing, and I know I've actually ahelped about issues before where I was trying to do something, it wouldn't work, and someone had to tell me that my intent was the issue.

Robustin commented on code/game/gamemodes/gang/vigilante.dm:
```diff
@@ -0,0 +1,203 @@
+	w_class = WEIGHT_CLASS_TINY
+	origin_tech = "programming=5;bluespace=2"
+	var/points = 0
+	var/list/tags = list()
```
Ive never made an entire list "lazy" or whatever and I can't see any examples that make it clear.

Robustin commented on code/game/gamemodes/gang/vigilante.dm:
```diff
@@ -0,0 +1,203 @@
+		return
+	playsound(get_turf(loc), 'sound/items/poster_being_created.ogg', 75, 1)
+	if(do_after(usr, 30, target = user))
+		if(I)
```
Yes but then it won't show the action bar (ive tested it). The in-hand action bars aren't working right now either.

Robustin commented on code/game/gamemodes/gang/gang_items.dm:
```diff
@@ -312,36 +568,147 @@
+	var/total_brute	= 0
+	H.notify_ghost_cloning("You're being injected with a revivification serum!")
+	if(do_after(user, 30, target = H))
+		if(H.stat == DEAD)
```
I poked around with this but couldn't quite get it working, do I have to make the checks into their own proc first?

Robustin commented on code/game/gamemodes/gang/vigilante.dm:
```diff
@@ -0,0 +1,203 @@
+	dat += "<a href='?src=\ref[src];choice=refresh'>Refresh</a><br>"
+
+	var/datum/browser/popup = new(user, "gangtool", "Welcome to Vigilante's Companion v1.1", 400, 750)
+	popup.set_content(dat)
```
Runtime in vigilante.dm,106: Cannot execute "Your Influence: 0<c...".Join().

FantasticFwoosh commented:
Being static would not make a mode like this or any "out of the box thinking" mode work, because it'd run into EXACTLY the same old faucets of security just deconverting everybody via processing them, deputized assistants and generally antags murdering innocents quite blatently.

It doesn't exclude anybody from the opportunity to be a antag or atleast morally grey with purpose.

Robustin committed:
Fixes and improvements and tweaks

Pascal125 commented:
You're being played.

Robustin committed:
Korcode

Robustin committed:
wew

Robustin committed:
Map memes ogre

Robustin committed:
Intro memes ogre

InsaneHyena commented:
If Robustin doesn't get to improve gang gamemode, I would rather not play gang at all. It's shit.

Robustin committed:
Explosion tweaks

Robustin commented:
Ok I think thats all of the intro bugs. Sec should be totally disabled, the explosions should go of all ALL maps, and bosses only get one gangtool.

Bawhoppen commented:
War ops has a redeeming factor of the entire crew cooperating and working together to prepare, even if it's in an unorthodox way

Robustin commented:
And at that point we're arguing about which TDM style is more enjoyable, not whether all-out armed conflict is a bad thing.

I really think that's where your opposition lies. Its crew fighting crew fighting crew.

I think, if people like this version, there's some weight we can take from Rev or >Wizard<, since the latter's summon rounds are about as low quality as you can get for a "TDM" round.

Robustin committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into GWTest

# Conflicts:
#	code/game/gamemodes/gang/gang_datum.dm
#	icons/obj/guns/projectile.dmi

Robustin committed:
More fixes, testmerge ready

optimumtact commented:
\>gang is so bad nobody wants to play sec
\>lets just remove sec instead of admitting the mode is shit

What is going on here? is this some kind of sunk cost fallacy? this shit is garbage, cut it loose.

ChangelingRain commented on code/game/gamemodes/gang/gang.dm:
```diff
@@ -25,10 +25,13 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
+	required_players = 0
+	required_enemies = 0
+	recommended_enemies = 0
+	enemy_minimum_age = 0
```
hoi

Robustin commented:
Where the fuck are you getting this "nobody wants to play sec" thing. Every argument for removing sec is "Sec shits on the whole aspect of Gang v. Gang and just turns it into another implant farming mode".

Robustin committed:
wew

Really-Good-Soda-Flavor commented:
You pretty much own the gamemode at this point so my complaints are moot, but I really think the whole blowing up security idea is stupid. It makes the round too chaotic and kills people who aren't security, too.

Limski commented:
Hi I don't know if I should be making an issue report for this but at the moment you can buy items from the  vigilante ability thing when you're unconscious/dead.

Robustin commented:
Hi Lim, I was in that round so I fixed it when you described it.

The biggest issue I'm having is this out-of-nowhere territory bug. I really can't figure out what's causing it since its not always replicating and even after adding debug messages to all territory additions I'm not seeing where its messing up. I'm going to have to start pestering someone else to help me investigate.

@Really-Good-Soda-Flavor 

Not moot, the mode itself will be dead if it can't find its groove and I don't take pride in making modes unfun.

The whole sec explosion thing will probably be a gap filler given the issues its had. If there's a clean way to delete all of the sec equipment I'll probably go that route (it might be nice for vigilantes or gangs to have a department thats relatively deserted to hole up in). 

Hopefully nonsec shouldn't be dying though, I disabled the lawyer for the sole reason that they were the only job close enough to be in the blast but maybe I overlooked a certain map.

Robustin committed:
fucking territory bug

Robustin committed:
Merge branch 'Future9' of https://github.com/Robustin/tgstation into GWTest

Robustin committed:
FINALLY FIXED

Robustin committed:
fuck gangcode

Robustin committed:
Test Ready

Robustin committed:
tweaks

Robustin commented:
Untested next version coming up. Reinforcement gateways have been tweaked and are available at roundstart for free. Gang bosses who attempt to place one will be warned that it will be their only source of reinforcements. 

Placing one early isn't going to get you an advantage, I'll have the first "reinforcement" not show up until a certain time in the round. But as the round goes on it will be a crucial source of fresh blood that needs to be kept safe. 

I've also given vigilantes access to influence-enhancing cleaning tools at roundstart but the soap is gone since I couldn't stomach the idea of that many bars of soap lying around. Vigilantes also get more periodic influence. I've also slowed down influence gain for bosses since its apparently that a decent gang could rack up 20+ tags in a matter of minutes and give the boss access to endgame weapons by the 10m mark - though im hoping vigilante cleaning at roundstart will also diminishing the ability for a gang to get an explosive start. I've also given a significant "mounting" time for the machine gun so it is no longer an effective way to "run and gun".

Robustin committed:
Significant changes, untested

AnturK committed:
Mapped active shieldgens

AnturK committed:
Makes VV'ing ID cards easier.

Mark9013100 committed:
[Omega]Fixes checkered plating

ChangelingRain committed:
Fixes ghosts using marker beacon

Bawhoppen commented:
this is completely the wrong idea on how to make gang work, and basically everyone agrees at this point

Iamgoofball commented:
![stop whining about tdm](http://vignette3.wikia.nocookie.net/trollpasta/images/f/f6/Alg-crying-baby-jpg.jpg/revision/latest?cb=20140323182823)

Bawhoppen commented:
Stop posting pics of yourself like that

Iamgoofball commented:
I post myself where I want

ChangelingRain commented:
Gang dominators are really, really weak to fire. Like, one guy with what was probably not an optimized flamethrower instantly kills a dominator weak.

Robustin committed:
ughhh

RemieRichards committed:
Advanced proccall now supports /proc/PROCNAME

RemieRichards committed:
If the comment were code, it would have errored!

Cyberboss committed:
Fixes posters

Robustin committed:
tested new version

Robustin committed:
Some improvements

Robustin committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Future94

Robustin committed:
More tweaks, testmerge ready

Robustin committed:
150 hours status = confirmed

Robustin committed:
Gud code

Robustin committed:
Why is Delta in my .dme?

MrsTonedOne committed:
kor is a butt (#27883)

ChangelingRain commented:
You can use your own advanced recruitment implant on yourself and it de-gangs you entirely

ChangelingRain commented:
Vigilantes lose their vigilante menu thing if cloned.

Tacolizard commented:
trying to reload the surplus rifle by using it makes it disappear, probably due to switching to a nonexistent sprite.

Tacolizard commented:
Gang bombs are super OP. They aren't hard to get and can wipe out an entire gang

Tacolizard commented:
Hovering over items in the influence menu should say what each item does. The amount of stuff is overwhelming

Tacolizard commented:
it should be slightly harder to convert people, vigilantes at least. It's weird being a vigilante for half the round only to have someone instantly convert you

Tacolizard commented:
it should be harder to tag and remove tags, to help make gang territory feel a bit more 'solid'. The spray noise should play during and before you start tagging/removing a tag

Tacolizard commented:
placing a dom should recall the shuttle

Tacolizard commented:
the dom is very weak to bullets. A couple hits and its ded. it may be a good idea to make the dom invincible

Commit:
Automatic changelog compile, [ci skip]

Tacolizard commented:
the mg turret needs a frontal shield or something. a grenade laucher/rpg is pretty much a perfect counter, and there's no way for the MG user to dodge or even really see what's coming

ChangelingRain commented:
The MG turret's bullets are kind of absurdly strong for how many it fires, though

VeryRealAccount commented:
This is even worse than it was before.  Turrets absolutely shred to the point your best method of attack is dragging a turret behind you, and influence is ridiculous for gangs while being near-impossible for Vigilantes.

Anonymous583 commented:
There is literally no point for Vigilantes to be in the game if they don't have an emergency win button. I've been sitting in this shitty Gang testing round for what feels like a century and at this point Death Squad should've been called a hundred times over. 

Fuck this game mode. Stop trying to make it work.

Tacolizard commented:
@VeryRealAccount You guys are super unrobust if you think getting influence as a vigilante is hard. I easily got a shotgun as a vig

VeryRealAccount commented:
I've never seen the admins actually spawn a Death Squad, I think they're just a meme at this point.  Complete breakdown of station order and all jobs essentially ceasing to exist as people instead fight over 'turf'?  Yeah, that's no reason at all to call in a Death Squad or even an ERT.

Tacolizard commented:
@Anonymous583 The admins aren't going to call in a deathsquad during a TEST. The purpose of the test is to see the entire round played through.

Tacolizard commented:
gang heads never make reinforcement portals.

PKPenguin321 commented:
Not sure this is the right way to go about this

Has this been testmerged

Bawhoppen commented:
>HAS THIS BEEN TESTMERGED

>BASICALLY 24/7 GANG TDM ALL DAY

>EVERYONE IN OOC COMPLAINS

Robustin commented:
@ChangelingRain 

I'm nerfing the MG in a couple ways but their bullets are literally among the weakest we have. It takes 8 shots to crit a full health gangster in their armored outfit. 

I'm trying to decide if Vigilantes are too strong now too. Either way I was kind of counting on Gang Leaders using the gateway now thats its FREE but it hasn't made an impact yet.

ChangelingRain commented:
Every time your pr gets testmerged I get large amounts of vocal negative feedback I am rating your PR a Mistake and closing it.
Democracy!

Iamgoofball commented:
So we can remove clock cult too right? The forum warriors hate that too.

duncathan commented:
(she was kidding when she said democracy)

ShizCalev committed:
Fixes APC (#27855)

optimumtact commented:
did a gang wipe out a clockcult or something?

Iamgoofball commented:
I'll be opening clock cult removal prs pending a poll.

Iamgoofball commented:
Thanks for clarifying what el prez meant Sean "@duncathan" Spicer

Robustin commented:
I can only point out that if "OOC shitposting" decided the fate of our modes then your clock cult would've been in the dirt a year ago. I expected something simple and fair like a server poll to decide if this was the right direction as I tweaked the value to address concerns. I figured someone like you would at least respect the need to tweak things in a mode to get them to a reasonable state.

I got one run for 2ish hours without major bugs before you decided enough was enough. The stunning hypocrisy is more aggravating to me than the actual closing of the PR.

vcordie commented:
we ran two server polls and testmerged it for several days. both forum posts and ooc has been in a total uproar, hating the mode. hating the core concepts. hating the general balance. hating the specific balance. hating literally every fucking thing you wrote. Just take the L and move on.

optimumtact commented:
It's a fair point, clockcult got a lot more time, @ChangelingRain unless you can come up with a convincing answer I'm tempted to reopen.

Robustin commented:
Its the most radical change to any mode that's ever been attempted besides the completely shit "Newcult" which I guarantee at something like 80% negative votes will always be our worst mode. 

Maybe I'm doomed to the same fate but I've been convinced by Kor and others to try this insane departure. Unfortunately it was Robustin(tm) code so the first two tests were significantly effected by huge bugs.

I didn't even want the 2nd merge to stick around. After someone spammed 100 gateways I figured it would be pulled but apparently it was left on for some unfathomable reason. Apparently the bulk of the tested rounds were done while I was offline making fixes without realizing an earlier iteration was still live.

It was only until 8pm today that we finally got the mode into something resembling normalcy with all of the major features functioning properly and working as intended. 

I believe we ran a total of 3 rounds across both servers during this window before you closed it up at around 11pm. I only saw 2 of the rounds and I know one basically had some shitbag maxcap everyone (I have TTV's out in my latest commit). 

I know there's been negative feedback, I'm addressing all of it. Vigilantes started too weak, Machinegun was overtuned. Despite having several ways to bring people back into the round, all were too expensive or too infrequent. This is a perfectly normal aspect of introducing what is effectively a new mode with a wide assort of new weapons and a balance of power (nosec) that we've never tried before. 

I expect to get about a week with a couple tests every night before we'd run some kind of poll and see if this is worth pursuing. Instead I get 3 hours with Vcordie screeching in OOC before every round. This was a bold experiment and knew failure was a possibility, I'm not that upset about it being closed, I'm just (as usual) more upset at the hypocrisy involved. Changelingrain gave herself MONTHS and DOZENS OF PR's to work on several unpopular modes and concepts (Holoparasites, Blob, Clock Cult) and at no point did negative reactions in OOC ever factor into her decisions. Yet a few hours with this PR finally working and OOC, the lowest fucking common denominator for feedback in this game, was enough to persuade you. We also had a 90 minute rev round that had most of the server screeching in OOC after, and god knows OOC just *LOVES* clock cult. Guess we should remove both of those as well. 

Is it really that hard to run this for a few more evenings while I try to undertake this enormous juggling act of making a no-security mode feasible and then hold a POLL on the server where people can offer their honest opinion about the future of the mode without worrying about Vcordie screeching at them in all caps. I've also seen plenty of posts supporting this change or at least open to seeing how it plays out with some appropriate nerfs/buffs.

Shadowlight213 committed:
Lowers portable pump max pressure to 1013.25

Bawhoppen commented:
why do we need insane over the top tdm instead of just a simple non completely disruptive round mode like I suggested

Robustin commented:
@Bawhoppen 
Because it can sit at 5% and offer variety without being crazy every round. Id rather have that over another mediocre sec stomps gang flavor that isn't disruptive but objectively awful.

@KorPhaeron 

This was a bold experiment and like so many things around here it will live or die based on your word. This PR got a pathetic amount of functional uptime for such a major change but if you don't support it anymore then there's no point in re-opening. I think this mode can earn a niche in its no-sec form. I would propose testing through this weekend as I address certain balance issues and then give the players the final say but that's just my preference.

Steelpoint commented:
I just think its utterly hypocritical that @ChangelingRain closed this PR due to a few bad rounds, that according to @Robustin were using out of date code, whereas we were forced to stick with the shittiest versions of Clock Cult and other game modes for weeks, if not months, until they finally got balanced into something a bit worthwhile. 

Completely shameless, if @ChangelingRain has any sense of integrity she would reopen this PR and grant Robustin at least a week of testing and a in-game poll.

AnturK committed:
You know i should probably add QDEL_LAZYLIST instead

lzimann commented:
Just pretend I closed the pr for the reason: lol.
There you go, no hipocrisy.

Jordie0608 committed:
Fixes faulty query and logic (#27890)

lzimann committed:
Merge pull request #27870 from ChangelingRain/altunclickable

Fixes ghosts using marker beacon

lzimann committed:
Merge pull request #27869 from Mark9013100/omega_plating_5/30/2017

[Omega] Fixes checkered plating

lzimann committed:
Merge pull request #27866 from AnturK/something-i-remembered

Makes VV'ing ID cards easier.

ChangelingRain committed:
Merge pull request #27864 from AnturK/more-mapper-stuff

Mapped active shieldgens

ShizCalev committed:
[Meta] Var Cleanup and tweaks (#27871)

Robustin commented:
Yea but you didn't because you got shut down before for unilaterally trying to remove the mode, trying to shut down the only person making changes to it a week later would've been more of the same drama.

But bring in a new maintainer to go "lol I saw 6 OOC shitposts about it, CLOSED" and its another story.

lzimann commented:
lol I saw 6 OOC shitposts about it, CLOSED

MrStonedOne commented:
robustin's right,

Robustin commented:
If there's no decisive action here at I'll play the meme game and open a forum poll. If suddenly OOC shitposters get to decide the fate of a mode then surely a ~formal poll~ will be enough to automerge my future Gang War PR's right? hah...

lzimann commented:
All you have to do is talk with Kor/Cheridan, if they say no, then well, though luck.

silicons committed:
Adds repair and lavaland map generator templates for buildmode use (#27865)

silicons committed:
Beam rifle combat rebalance (#27867)

Expletive committed:
Adds The NT-75 Electromagnetic Power Inducer (#27653)

tgstation-server committed:
Automatic changelog generation for PR #27653 [ci skip]

lzimann committed:
Fixes master

lzimann committed:
Merge pull request #27911 from lzimann/ree

Fixes master

Bawhoppen commented:
I almost feel bad for robustin at this point

silicons committed:
Gang Machinegun Mouse Tracking + Projectile Source Fix (#27868)

Moves Gang Machineguns to its own file. It's not a goddamn portable turret and it's not even something that uses power!

Gang Machineguns now track the mouse location of the first buckled person (For those who don't code, that means the person using it)

Gang Machineguns will automatically aim and turn towards whatever that person is pointing at.. 
However, to make it target something specifically instead of a direction, you still have to hold your mouse down while firing.

Gang machineguns will no longer cause the user to shit out bullets if they unbuckle during firing :^)

SOME REFACTORS TO PIXEL PROJECTILES ARE INCLUDED...

tgstation-server committed:
Automatic changelog generation for PR #27868 [ci skip]

optimumtact committed:
Merge pull request #27877 from tgstation/RemieRichards-patch-1

Advanced proccall now supports /proc/PROCNAME

optimumtact committed:
Merge pull request #27879 from Cyberboss/CultFix

Fixes posters

Shadowlight213 committed:
Borg module resets now require cutting instead of pulsing the reset wire (#27880)

tgstation-server committed:
Automatic changelog generation for PR #27880 [ci skip]

MMMiracles committed:
CereStation Security Department Revamp (and other misc fixes/changes) (#27884)

tgstation-server committed:
Automatic changelog generation for PR #27884 [ci skip]

Steelpoint committed:
InitialCommit (#27908)

tgstation-server committed:
Automatic changelog generation for PR #27908 [ci skip]

Robustin commented:
Kor was confident that someone else would unlock it. Either way if nobody else cares enough to give this a shot I'll wait for a forum poll to provide enough Joan OOC meme repellent before I waste any more time.

lzimann committed:
Merge pull request #27894 from Shadowlight213/idon'trememberwhyleowantedmetodothis

Lowers portable pump max pressure to 1013.25

tgstation-server committed:
Automatic changelog generation for PR #27894 [ci skip]

optimumtact commented:
herp

optimumtact commented:
gang is garbage, remove the mode

Robustin commented:
You people are going to be the end of me.

I'll give the numbers another pass, hopefully I didn't lose the unpushed commits I made, and I'll leave security unexploded and just search+qdel the problematic equipment during setup. 

The biggest thing I want to stress is that I WANT this to be a niche mode. If I had my way I would give this mode's 3.5% back to whatever we took it from and cut wizard in half, remove summon guns, and have this mode make up the 4.75% that was cut from Wizard. Unless it succumbs to memes again it'll hopefully be ready for another test during the next primetime, after that I'll open a forum poll and see where people lean on NoSec v. Sec Gang War, whether people want the emphasis on slower rounds with base-building aspects or faster rounds that will typically wrap up within 30m.

Robustin committed:
Unpushed changes

Goodstuff123456 committed:
Fixes black carpet recipe and sprite (#27891)

tgstation-server committed:
Automatic changelog generation for PR #27891 [ci skip]

Cyberboss committed:
Moves tgstation-server (#27904)

* Removes tgstation-server

* Add the server submodule

* Bring it up to date

lzimann committed:
Fixes show all logs runtiming, also allows multiple individual log windows open (#27903)

Robustin committed:
Merge branch 'Future9' of https://github.com/Robustin/tgstation into Future96

# Conflicts:
#	code/game/machinery/porta_turret/portable_turret.dm
#	icons/mob/inhands/items_lefthand.dmi
#	icons/mob/inhands/items_righthand.dmi
#	tools/tgstation-server

Robustin committed:
Actually some semblance of balance edition

PKPenguin321 commented:
okay so politics of this pr itself being opened or closed, i'm still not sure this is the right way to tackle gang. this is moving away from improvement to outright changing fundamental playstyles behind the mode. it does a lot at once and just seems like it's not gang anymore. and i know lore shouldn't come into account when it comes to gameplay, but reviving dead gang members with a purchasable power? come on. this is supposed to be street gang styled. at this point it may as well be themed after two futuristic army factions fighting over the station instead of gangs or whatever.

since youve said this is a test, please dont expect it to go through. im open to trying it but i just dont think this is right at the moment.

Robustin commented:
I mean I kept almost everything low-tech themed, surplus rifle, improvised shotgun, pitchfork, shuriken, etc. are still the staples of combat and only at the end does some higher tech stuff become feasisble. 

The revive is themed as essentially an adrenaline shot with some unspecified banned drugs mixed in and the description and visual/textual feedback definitely give the "this is not advanced technology" vibe.

It does change a lot from classic gang but Kor was giving me feedback on where he thinks the mode might survive, you've been silent. I know something that shakes things up this much is far from a guaranteed merge though. It's ready for the next round of testing though.

Robustin commented:
Wait what the hell happened with the commits this is literally the upstream master from a few hours ago with this branch pulled onto it  why is it fucking with the maps and shit?

Robustin commented:
Memes
